contrib/jinzhu/gorm/option.go

```diff
@@ -74,3 +76,11 @@ func WithCustomTag(tag string, tagFn func(scope *gorm.Scope) interface{}) Option
 		cfg.tagFns[tag] = tagFn
 	}
 }
+
+// WithErrorCheck specifies a function fn which reports whether the passed
+// err should be considered an error.
```

Comment:
We should be more clear about what this is for. I suggest something like:
 > WithErrorCheck specifies a function fn which determines whether the passed error should be marked as an error. The fn is called whenever a gorm operation finishes with an error